### DISCUSSION TITLE
Stream sends code to screen

### DISCUSSION THREAD

**Discussion by MarcPinnell** [dompdf/dompdf#3053] [Q&A] [answered]
I am trying to generate a pdf, save it to the server and then stream the same generated PDF to the screen. The following code (at the bottom of the PDF generation (which works fine) saves the file/PDF and then sends raw PDF code to the screen (like a text file):

```
		// $file = $path.'Consignment_Original.pdf';
		file_put_contents($fileName.'.pdf', $dompdf->output());

		// Output the generated PDF to Browser
		$dompdf->stream($fileName,array("Attachment"=>0));

		return;
```

This code however works fine, but my PDF generator is in a function and I need to return to the original block of code, so it doesn't work from a program standpoint.

```
		// $file = $path.'Consignment_Original.pdf';
		file_put_contents($fileName.'.pdf', $dompdf->output());

		// Output the generated PDF to Browser
		$dompdf->stream($fileName,array("Attachment"=>0));

		exit();
```

Any suggestions?

**Comment by bsweeney** [comment, marked as answer]
There's a known bug in the "compilation" processes in the Cpdf library that break when called twice. To avoid the issue you should use only the `output` or the `stream` method, but not both. Since you're saving the file to disk you should be able to stream that file to the browser using [readfile](https://www.php.net/manual/en/function.readfile.php). Just replace the stream call with something like the following (tweaked from the PHP man page):

```
header('Content-Description: File Transfer');
header('Content-Type: application/pdf');
header('Content-Disposition: attachment; filename="'.basename($fileName).'"');
header('Expires: 0');
header('Cache-Control: must-revalidate');
header('Pragma: public');
header('Content-Length: ' . filesize($fileName));
readfile($fileName);
exit;
```

**Reply by MarcPinnell** [reply]
Thanks
